src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -60,6 +59,8 @@ public struct MetricPoint
                 this.bucketCounts = null;
                 this.lockObject = null;
             }
+
+            this.StartTime = startTime;
```

Comment:
moved StartTime assignment to the very end to make sure when startTime is assigned, struct is fully assigned. But this is likely incorrect as DateTimeOffset is a struct with multiple fields, so assignment is not atomic.. 
Another option is to store the Ticks, which is long, and then leverage Interlocked on log...